clang/lib/DPCT/DPCT.cpp

```diff
@@ -498,6 +498,62 @@ static void loadMainSrcFileInfo(clang::tooling::UnifiedPath OutRoot) {
   }
 }
 
+std::vector<std::string> SplitStr(const std::string &Str, char Delimiter) {
+  std::vector<std::string> Result;
+  std::istringstream Stream(Str);
+  std::string Token;
+  while (std::getline(Stream, Token, Delimiter)) {
+    Result.push_back(Token);
+  }
+  return Result;
+}
+
+bool IsTookitInstalledOnTargetOS(std::map<std::string, bool> &AvaliableTool) {
+  const char *EnvPath = std::getenv("PATH");
+  bool Ret = false;
+  if (EnvPath == nullptr) {
+    return Ret;
+  }
+  std::string EnvPathStr(EnvPath);
+  std::vector<std::string> PathDir = SplitStr(EnvPathStr,
+#if defined(_WIN32)
+                                              ';'
+#else
+                                              ':'
+#endif
+  );
+  for (const auto &Dir : PathDir) {
+    for (auto &Tool : AvaliableTool) {
+      std::string ToolPath = appendPath(Dir, Tool.first);
+      if (llvm::sys::fs::exists(ToolPath)) {
+        Tool.second = true;
+        return true;
+      }
+    }
+  }
+  return Ret;
+}
+
+std::string GetInstalledPython() {
+  std::map<std::string, bool> Tool = {
```

Comment:
Can we just use a `unordered_set` and return the exist one in `IsTookitInstalledOnTargetOS`. Then we need not the `for` loop in line 548.

Reply:
Update